app/src/main/java/com/github/weg_li_android/MainActivity.kt

```diff
@@ -25,12 +59,177 @@ class MainActivity : AppCompatActivity() {
             ViewModelFactory(ApiHelper(ApiServiceImpl()))
         ).get(MainViewModel::class.java)
 
+        setupPhotoCard(this)
+
         setupCarTypeSpinner()
         setupViolationSpinner()
         durationText.setOnClickListener {
         }
 
         sendButton.setOnClickListener { mainViewModel.sendReport() }
+
+        val licenseObserver = Observer<Report> { newName ->
+            if(newName.license != "") {
+                status_image.setImageResource(R.drawable.ic_baseline_check_circle_24)
+            }
+            else {
+                status_image.setImageResource(R.drawable.ic_baseline_error_24)
+            }
+        }
+
+        mainViewModel.propertyAwareReport.observe(this, licenseObserver)
+
+        car_license_plate_edit_text.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(
+                s: CharSequence,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
+            }
+
+            override fun onTextChanged(
+                s: CharSequence,
+                start: Int,
+                before: Int,
+                count: Int
+            ) {
+            }
+
+            override fun afterTextChanged(s: Editable) {
+                mainViewModel.changeLicense(s.toString())
+            }
+        })
+    }
+
+    private fun setupPhotoCard(context : Context) {
+        setupPhotoRecyclerView(context)
+
+        val pm: PackageManager = context.packageManager
+        if (!pm.hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
+            take_picture_button.visibility = View.GONE
+        }
+
+        val photosCountObserver = Observer<Report> { newName ->
+            if(newName.violationPhotos.size >= 2) {
+                photos_status_image.setImageResource(R.drawable.ic_baseline_check_circle_24)
+            }
+            else {
+                photos_status_image.setImageResource(R.drawable.ic_baseline_error_24)
+            }
+        }
+
+        mainViewModel.propertyAwareReport.observe(this, photosCountObserver)
+    }
+
+    fun convertPixelsToDp(px: Float, context: Context): Float {
+        return px / (context.resources
+            .displayMetrics.densityDpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)
+    }
+
+    private fun Float.convertDpToPixel(context: Context): Float {
+        return this * (context.resources
+            .displayMetrics.densityDpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)
+    }
+
+
+    private fun setupPhotoRecyclerView(context: Context) {
+        val recyclerView = findViewById<RecyclerView>(R.id.photos_grid)
```

Comment:
Please either use kotlin synthetics or the new View Binding (which afaik is now the recommended way to do this), see https://medium.com/androiddevelopers/use-view-binding-to-replace-findviewbyid-c83942471fc

Reply:
I would just use "photos_grid" (should be renamed to "photosRecyclerView"), kotlin will do the rest.